components/eamxx/src/share/util/eamxx_time_interpolation.cpp

```diff
@@ -0,0 +1,343 @@
+#include "share/util/eamxx_time_interpolation.hpp"
+
+namespace scream{
+namespace util {
+
+/*-----------------------------------------------------------------------------------------------*/
+// Constructors
+TimeInterpolation::TimeInterpolation(
+  const grid_ptr_type& grid 
+)
+{
+  // Given the grid initialize field managers to store interpolation data
+  m_fm_time0 = std::make_shared<FieldManager>(grid);
+  m_fm_time1 = std::make_shared<FieldManager>(grid);
+  m_fm_time0->registration_begins();
+  m_fm_time0->registration_ends();
+  m_fm_time1->registration_begins();
+  m_fm_time1->registration_ends();
+}
+/*-----------------------------------------------------------------------------------------------*/
+TimeInterpolation::TimeInterpolation(
+  const grid_ptr_type& grid, 
+  const vos_type& list_of_files
+) : TimeInterpolation(grid)
+{
+  set_file_data_triplets(list_of_files);
+}
+/*-----------------------------------------------------------------------------------------------*/
+/* A function to perform time interpolation using data from all the fields stored in the local
+ * field managers.
+ * Conducts a simple linear interpolation between two points using
+ *        y* = w*y0 + (1-w)*y1
+ * where w = (t1-dt)/(t1-t0)
```

Comment:
Minor: seeing `dt` in the comment seems confusing. That's not a timestep, but a _timestamp_. So maybe `w=(t1-t) / (t1-t0)`?